### PR TITLE
Add Viem Docs

### PR DESCRIPTION
**Pull request by themacexpert**
### Description

Add Viem Docs

### Checklist

- [x] I have added a label to this PR 🏷️
- [x] If this page requires a disclaimer, I have added one
- [ ] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects

### CODE REVIEW

File: builders/interact/ethereum-api/libraries/viem.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    --8<-- 'code/ethereum-api/viem/transaction.ts'
+    ```
+
+To run the script, you can run the following command in your terminal:
+
+```bash
+npx ts-node transaction.ts
+```
+
+If the transaction was successful, in your terminal you'll see the transaction hash has been printed out. You can also use the `balances.ts` script to check that the balances for the origin and receiving accounts have changed. The entire workflow would look like this:
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/viem.md:169]:
```suggestion
If the transaction was successful, in your terminal, you'll see the transaction hash has been printed out. You can also use the `balances.ts` script to check that the balances for the origin and receiving accounts have changed. The entire workflow would look like this:
```

File: builders/interact/ethereum-api/libraries/viem.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+![The result of running the transaction and balances scripts in the terminal](/images/builders/interact/ethereum-api/viem/viem-2.png)
+
+## Deploy a Contract {: #deploy-contract }
+
+--8<-- 'text/libraries/contract.md'
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/viem.md:175]: This snippet has a grammar issue: "Next you can add the Solidity code to the file:" should be "Next, you..."

File: .snippets/code/ethereum-api/viem/.prettierrc.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// prettierrc.cjs or .prettierrc.js
```
Comment (eshaben) [inline .snippets/code/ethereum-api/viem/.prettierrc.js:1]: This doesn't look like this file is used anywhere, and we typically don't tell users how to configure prettier in any of our demos anyways, so I think this file can probably be deleted

File: .snippets/code/ethereum-api/viem/balances.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
+// 1. Imports the necessary components from viem
```
Comment (eshaben) [inline .snippets/code/ethereum-api/viem/balances.ts:1]: This should just be "Import" not "Imports", please check all the snippets

File: .snippets/code/ethereum-api/viem/deploy.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+  // 7. Send tx (initial value set to 5)
+  const contract = await walletClient.deployContract({
+    abi,
+    account,
+    bytecode,
+    args: [_initialNumber],
+  });
+
+  // 8. Get the transaction receipt for the deployment
```
Comment (eshaben) [inline .snippets/code/ethereum-api/viem/deploy.ts:72]: Step 8 spells out "transaction" and step 7 doesn't. Let's just use "transaction". Please check all snippets

File: .snippets/code/ethereum-api/viem/read-chain-data.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
+// 1. Import the necessary components from viem.
```
Comment (eshaben) [inline .snippets/code/ethereum-api/viem/read-chain-data.ts:1]:
```suggestion
// 1. Import the necessary components from viem
```

File: builders/interact/ethereum-api/libraries/viem.md [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+4. Define the `addressFrom` and `addressTo` variables
+5. Create the asynchronous `balances` function that wraps the `publicClient.getBalance` method
+6. Use the `publicClient.getBalance` function to fetch the balances for the `addressFrom` and `addressTo` addresses. You can also leverage the `formatEther` function to transform the balance into a more readable number (in UNIT)
+7. Lastly, run the `balances` function
+
+??? code "View balances.ts"
+
+    ```ts
+    --8<-- 'code/ethereum-api/viem/balances.ts'
+    ```
```
[the remark this answers is not in the document]
Comment (eshaben) [reply builders/interact/ethereum-api/libraries/viem.md:125]: Ah ok, I see where maybe this went a little off the tracks. If you're trying to pull in the creating the client part, you can reference the Ethers tutorial where we use `// {...}` this type of comment to indicate pulling in the other parts and then use the expandable snippet to tie it altogether

[the remark this answers is not in the document]
Comment (eshaben) [reply builders/interact/ethereum-api/libraries/viem.md:125]: either way, we should be displaying a visible code snippet for the users at the end of the steps

**Review by eshaben — CHANGES_REQUESTED** [review]
nice job! the index page icons look a bit pixelated/blurry and they don't look like the right size. are you using an svg or png?

**Comment by themacexpert** [comment]
I made all snippets expanded by default so it looks longer but it corresponds nicely to the #1-9 steps described in the tutorial. If it seems too big we can re-visit another approach.

File: builders/interact/ethereum-api/libraries/viem.md [hunk 8]
```diff
@@ -0,0 +1,322 @@
...
+```
+
+### For Writing Chain Data {: #for-writing-chain-data }
+
+To create a client for writing chain data, you can take the following steps:
+
+1. Import the `createWalletClient`, `http`, and `defineChain` functions from `viem`, and the `privateKeyToAccount` function from `viem/accounts`
+2. Define the chain details of your EVM ContainerChain, making sure to include all fields shown below. Both `public` and `default` RPC URLs are required to be listed, even if they are the same
+3. Create your account using the `privateKeyToAccount` function
+4. Create the `client` using the `createWalletClient` function and pass in the account, network, and the HTTP RPC endpoint
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/viem.md:80]: There are 4 steps here, but only 3 listed in the snippet

**Review by eshaben — COMMENTED** [review]
I think the expandables look great! 🙌 

The viem icons need to fill the width or height (whichever is greater) of the 256 x 256px background. It looks like the logo is also a png because its a bit pixelated, but I could be wrong! I can find an svg if you need one! 🙂

**Comment by themacexpert** [comment]
Thanks @eshaben, fixed!

**Review by eshaben — APPROVED** [review]
LGTM! 🚢 ✨

**Pull request merged**